src/civilite/schedule.py

```diff
@@ -50,11 +51,21 @@ def __init__(self, start: time, stop: time) -> None:
 class WeeklySchedule:
     '''A collection of events that repeat every week.'''
 
-    def __init__(self) -> None:
+    def __init__(self, observer: Observer, tzinfo: pytz.tzinfo = None) -> None:
+        # The Observer associated with this schedule.
+        self.observer = observer
+        if tzinfo is None:
+            # Use the timezone at the observer's location
+            tzf = TimezoneFinder()
+            self.tzinfo = pytz.timezone(tzf.timezone_at(
+                lat=observer.latitude, lng=observer.longitude))
+        self.tzinfo = tzinfo
```

Comment:
should this be in an else? seems like if tzinfo is one, we set self.tzinfo, then the next line resets it to None. 

Is there a test for this? I'll look maybe I can write one

Reply:
Ha! You're right, I made a bug.  But after some investigation, I found something interesting..

In this context, calls like `astral.sun.twilight()` result in astral calling `.astimezone(tzinfo)` on the given datetime object.  When `tzinfo` is `None`, the call effectively becomes something like:

`>>> pytz.utc.localize(datetime.datetime.utcnow()).astimezone(None)`

which returns:

`datetime.datetime(2020, 5, 9, 1, 25, 52, 659647, tzinfo=datetime.timezone(datetime.timedelta(days=-1, seconds=72000), 'Eastern Daylight Time'))`

According to [Python docs](https://docs.python.org/3/library/datetime.html#datetime.datetime.astimezone) on `datetime.astimezone()`, it picks up the timezone from the OS in that case:

"If called without arguments (or with `tz=None`) the system local timezone is assumed for the target timezone. The `.tzinfo` attribute of the converted datetime instance will be set to an instance of timezone with the zone name and offset obtained from the OS."

That's why it still produces the correct **sunsets.csv**.

Given this development, we don't need `timezonefinder` or its friend `numpy`.  I think relying on the correct timezone/locale setup in a client's OS is faster and more reliable.

This means I can remove the entire `if` block and remove `timezonefinder` from the project.

There's no unit test for this, you should write one.  My only quick test during development was to change lat/lng values significantly and then wondering why **sunsets.csv** was still the same.

Reply:
Another thought: we could follow astral's convention and change the default `tzinfo` to `pytz.utc`.  Then a caller would have to explicitly pass `None` if the local system timezone is desired.